src/widgets/favoritewidget.cpp

```diff
@@ -50,12 +50,16 @@ QSize FavoriteWidget::sizeHint() const {
 }
 
 void FavoriteWidget::paintEvent(QPaintEvent* e) {
-  QStylePainter p(this);
+  // ignore text descent when vertically centering
+  QFontMetrics fontMetrics = QFontMetrics(this->font());
+  QRect rect = this->rect();
+  rect.setBottom(rect.bottom() - fontMetrics.descent());
```

Comment:
If you put this code here, it will be called whenever paintEvent gets called, which is unnecessary. why not move this to the constructor ?